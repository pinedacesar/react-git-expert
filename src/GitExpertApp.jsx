import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GitExpertApp = () => {
  const [categories, setCategories] = useState(['Los Simpson']);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    // console.log(onNewCategory);
    // console.log('Valorant');
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GitExpertApp</h1>

      <AddCategory
        // setCategories={setCategories}
        onNewCategory={onAddCategory}
      />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
      {/* <li>ABC</li> */}
    </>
  );
};
