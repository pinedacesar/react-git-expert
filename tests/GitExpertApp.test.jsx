import { render, screen } from '@testing-library/react';
import { GitExpertApp } from '../src/GitExpertApp';

describe('Prueba de <GitExpertApp/>', () => {
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GitExpertApp />);
    expect(container).toMatchSnapshot();
    screen.debug();
  });

  // test('verificar si las categorias son iguales', () => {

  //   render(<GitExpertApp />);
  // });
});
