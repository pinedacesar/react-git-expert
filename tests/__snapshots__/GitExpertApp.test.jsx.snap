// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GitExpertApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GitExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Los Simpson
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
